tests/python/gaia-ui-tests/gaiatest/apps/system/regions/cards_view.py

```diff
@@ -72,6 +73,9 @@ def close_app(self, app):
         self.wait_for_card_ready(app)
         self.marionette.find_element(*self._close_button_locator(app)).tap()
         self.wait_for_element_not_present(*self._app_card_locator(app))
+        if len(self.cards) == 0:
+            self.marionette.switch_to_frame()
+            self.wait_for_condition(lambda m: self.apps.displayed_app.name == Homescreen.name)
```

Comment:
I'll make the change, but I'm not sure I agree with this.
I'd like to see these kinds of methods (in the app/pageregion objects) as something that's doing a UI action or a programmatic action and then it would make sure that it's in a stable state afterwards.
In case there are no cards left with the close_app method, the device returns to the Homescreen app. In my opinion, this method should wait for that state to happen.